File: src/dotnet/Domain/Day1/Day1.cs
namespace Domain.Day1;

public class Day1
{
    public static int GetsMaxCalories(IEnumerable<string> calories)
    {
        var maxCalories = calories
            .Select(num => num
                .Split("\n")
                .Select(n => Convert.ToInt32(n))
                .Sum()
            ).Max();

        return maxCalories;
    }

    public static int GetsSumOfTopThree(IEnumerable<string> calories)
    {
        var sumOfTopThree = calories
            .Select(num => num.Split("\n").Select(n => Convert.ToInt32(n)).Sum()
            ).OrderByDescending(n => n)
            .Take(3)
            .Sum();

        return sumOfTopThree;
    }
}

File: src/python/domain/day1/day1.py
class Day1:

    @staticmethod
    def get_highest_calories(calories: [str]):

        totalCalories = [sum(map(int, calorie.split('\n'))) for calorie in calories]

        return max(totalCalories)

    @staticmethod
    def get_top3_calories(calories: [str]):

        totalCalories = sorted([sum(map(int, calorie.split('\n'))) for calorie in calories], reverse=True)

        return sum(totalCalories[:3])
